app.py

```diff
@@ -180,23 +180,22 @@ def sample_ballots(election, round):
 
     batch_sizes = {}
     batches_to_ballots = {}
-    seen_ballot_positions = set()
     # Build batch - batch_size map
-    for batch_id, ballot_position in sample:
+    for item in sample:
+        batch_id, ballot_position = item[1] 
+        times_sampled = item[2]
+        ticket_number = item[0]
 
         lookup = (batch_id, ballot_position)
-        # Only count ballots once here since it's only pulled once
-        if lookup in seen_ballot_positions:
-            batches_to_ballots[batch_id].append(ballot_position)
-            continue
 
-        seen_ballot_positions.add(lookup)
         if batch_id in batch_sizes:
-            batch_sizes[batch_id] += 1
-            batches_to_ballots[batch_id].append(ballot_position)
+            if times_sampled == 1: # since we've already seen it, it doesn't affect batch size
+                batch_sizes[batch_id] += 1
```

Comment:
since we're now recording each time the ballot is sampled as separate row, shouldn't we kill the `times_sampled` field?

Reply:
This is just for bookkeeping for balanced retrieval lists. I could modify that to compute the batch sizes, but it'll take a little more re-engineering.

Reply:
I'm worried the SQL data model can become inconsistent, e.g. what if `times_sampled` says 3, but there are actually 4 matching `SampledBallot` rows? Also, are we updating all the `times_sampled` fields for all the matching sampled ballots?

Reply:
This field is never used outside of this function, so I'm not really sure how inconsistency would factor into it? This is just a back-of-the-envelope to figure out how many pieces of paper are in each batch so we can balance workload. All we need to know is that the ballot has been already sampled.

Reply:
@benadida if you'd rather me read a value out of the DB for that, then I can, but that seems like a lot of overhead for a thing the sample is already giving us anyway.